lib/task/import/importDipObjectsTask.class.php

```diff
@@ -0,0 +1,585 @@
+    // extension removed) with original object filename (example:
+    // 979c4458-21f3-11e1-a4bd-001d09282b9d-foo.tif)
+    $filenames = $this->createFilenameLookup($objectsPath);
+
+    // Determine column order and what type of unique value will be used to find
+    // information objects (slug or identifier)
+    $this->processCsvHeaderRow($fh);
+
+    $digitalObjects = array();
+
+    // Build hash on information_object key, with value being an array if information
+    // object has multiple digital objects attached
+    while ($row = fgetcsv($fh, 1000))
+    {
+      $filename = $this->getRowColumnValue('filename', $row);
+      $filepath = rtrim($objectsPath, '/') .'/'. $filename;
+
+      // Check if filename has been changed by Archivematica
+      if (!file_exists($filepath))
+      {
+        $key = null;
+
+        // Substitute original file extension (e.g. TIFF) with new DIP extension
+        // (e.g. JPEG)
+        if (preg_match('/(.+)\.(\w{3})$/', $filename, $matches))
+        {
+          $key = strtolower($matches[1]);
+        }
+
+        if (isset($key) && isset($filenames[$key]))
+        {
+          $filepath = str_replace($filename, $filenames[$key], $filepath);
+        }
+        else
+        {
+          throw new Exception("Error: Couldn't find file $filepath");
+        }
+      }
+
+      $uniqueValue = $this->getRowColumnValue($this->uniqueValueColumnName, $row);
+
+      if (!isset($digitalObjects[$uniqueValue]))
+      {
+        $digitalObjects[$uniqueValue] = $filepath;
+      }
+      else if (!is_array($digitalObjects[$uniqueValue]))
+      {
+        $digitalObjects[$uniqueValue] = array($digitalObjects[$uniqueValue], $filepath);
+      }
+      else
+      {
+        $digitalObjects[$uniqueValue][] = $filepath;
+      }
+    }
+
+    return $digitalObjects;
+  }
+
+  /**
+   * Determine CSV column order and which information object attribute will be
+   * used as a unique value to specify information objects to import digital
+   * objects to.
+   *
+   * @param resource $fh  File pointer resource for CSV file
+   *
+   * @return void
+   */
+  protected function processCsvHeaderRow($fh)
+  {
+    // Storage order of columns in CSV header row
+    $this->columnNames = fgetcsv($fh, 1000);
+
+    // Make sure there isn't both an "identifier" and "slug" column in the CSV header row
+    if (in_array('identifier', $this->columnNames) && in_array('slug', $this->columnNames))
+    {
+      throw new sfException('Error: CSV header row includes both an "identifier" column and a "slug" column. Please use only one of the two.');
+    }
+
+    // Make sure there is either an "identifier" or a "slug" column  in the CSV header row
+    if (!in_array('identifier', $this->columnNames) && !in_array('slug', $this->columnNames))
+    {
+      throw new sfException('Error: CSV header row must include either an "identifier" column or a "slug" column.');
+    }
+
```

Comment:
We could make both 'in_array()' checks at the begining of the function and asign the values to variables to avoid making them again other times.